src/daemon/command_parser_executor.cpp

```diff
@@ -334,17 +334,17 @@ bool t_command_parser_executor::set_limit(const std::vector<std::string>& args)
   if(args.size()==0) {
     return m_executor.get_limit();
   }
-  int limit;
+  int64_t limit;
   try {
-      limit = std::stoi(args[0]);
+      limit = std::stoll(args[0]);
   }
   catch(std::invalid_argument& ex) {
       return false;
   }
-  if (limit==-1)  limit=128;
-  limit *= 1024;
+  if (limit > 0)
+    limit *= 1024;
```

Comment:
If the limit is, eg, -10, this'll end up setting a huge (unsigned) limit once passed to the RPC.

Reply:
No, the RPC will error with `CORE_RPC_ERROR_CODE_WRONG_PARAM`

Reply:
Oh, I thought I'd seen a !=1, but it's != -1 as I look now, makes sense. Guess I should have triple checked :D